src/components/forms/cbr/CbrZoneExclusionAddressForm.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../../component-utils";
+import { IcseFormGroup } from "../../Utils";
+import { IcseNameInput, IcseTextInput } from "../../Inputs";
+import { IcseSelect } from "../../Dropdowns";
+import { cbrInvalid } from "../../../lib/cbr-utils";
+
+const typeNameMap = {
+  ipAddress: "IP Address",
+  ipRange: "IP Range",
+  subnet: "Subnet",
+  vpc: "VPC",
+  serviceRef: "Service Ref",
+};
+
+/**
+ * Context-based restriction addresses / exclusions
+ */
+class CbrZoneExclusionAddressForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    buildFormDefaultInputMethods(this);
+    buildFormFunctions(this);
+  }
+
+  handleInputChange(event) {
+    let { name, value } = event.target;
+    if (name === "type")
+      value = Object.keys(typeNameMap).find(
+        (key) => typeNameMap[key] === value
+      );
+    this.setState({ [name]: value });
+  }
+
+  render() {
+    return (
+      <>
+        <IcseFormGroup>
+          <IcseNameInput
+            id={this.props.data.name + "-cbr-address-exclusion"}
+            componentName={this.props.data.name + "-cbr-zone"}
+            className={"fieldWidthSmaller"}
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() =>
+              this.props.invalidCallback(this.state, this.props)
+            }
+            invalidText={this.props.invalidTextCallback(this.state, this.props)}
+            hideHelperText={true}
+          />
+          <IcseTextInput
+            id={this.props.data.name + "-cbr-account-id"}
+            componentName={this.props.data.name + "-cbr-zone"}
+            field={"account_id"}
+            value={this.state.account_id}
+            labelText={"Account ID"}
+            onChange={this.handleInputChange}
+            hideHelperText={true}
+            className="fieldWidthSmaller"
+            {...cbrInvalid("account_id", this.state.account_id)}
+          />
+          <IcseTextInput
+            id={this.props.data.name + "-cbr-zone-location"}
+            componentName={this.props.data.name + "cbr-zone-location"}
+            className={"fieldWidthSmaller"}
+            labelText={"Location"}
+            field="location"
```

Comment:
should this be a dropdown?

Reply:
Not sure - it's an optional field according to the provider, not sure how we
 would allow this with a dropdown, but i agree it makes more sense as a dropdown

<img width="652" alt="Screen Shot 2023-05-05 at 8 52 02 AM" src="https://user-images.githubusercontent.com/51029454/236476829-4dd43bde-a283-449c-b30e-53c44ba48414.png">

Reply:
what is `location` here?

Reply:
not entirely sure - there isn't an option for location even on the page when you create a zone through the console
<img width="969" alt="Screen Shot 2023-05-05 at 10 03 43 AM" src="https://user-images.githubusercontent.com/51029454/236495566-1fb5dfc0-c8a5-4b55-8216-9699f1e72daf.png">


perhaps it's referencing the location where a service instance is?

Reply:
let's just keep it as is then ◡̈